server/src/main/java/org/elasticsearch/action/admin/indices/flush/SyncedFlushResponse.java

```diff
@@ -124,6 +156,160 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static SyncedFlushResponse fromXContent(XContentParser parser) throws IOException {
+        ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+        parser.nextToken();
+        return innerFromXContent(parser);
+    }
+
+    private static SyncedFlushResponse innerFromXContent(XContentParser parser) throws IOException {
+        ShardCounts totalShardCounts = null;
+        Map<String, ShardCounts> shardsCountsPerIndex = new HashMap<>();
+        Map<String, List<ShardsSyncedFlushResult>> shardsResultPerIndex = new HashMap<>();
+        // If it is an object we try to parse it for Fields._SHARD or for an index entry
+        for (Token curToken = parser.currentToken(); curToken != Token.END_OBJECT; curToken = parser.nextToken()) {
+            ensureExpectedToken(Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
+            String fieldName = parser.currentName();
+            curToken = parser.nextToken();
+            Integer totalShards = null;
+            Integer successfulShards = null;
+            Integer failedShards = null;
+            Map<ShardId, List<FailureContainer>> failures = new HashMap<>();
+            if (curToken == Token.START_OBJECT) { // Start parsing for _shard or for index
```

Comment:
This is indeed true. Which is why I agree with your initial idea. It would be much better to have a separate response object altogether as this whole effort felt more like a workaround to me.

Reply:
we can't change the format of the response at this time. We can at some point, but for now we just have to parse what we have, and figure out what we should do to make things better for the future, meaning potentially breaking changes etc.